src/Rules.Framework/Core/IConditionNode.cs

```diff
@@ -11,6 +13,11 @@ public interface IConditionNode<TConditionType>
         /// </summary>
         LogicalOperators LogicalOperator { get; }
 
+        /// <summary>
+        /// Gets the condition node properties.
+        /// </summary>
+        IDictionary<string, object> Properties { get; }
```

Comment:
If we do that, then properties are focused only on the compilation 🙂 
The idea here was to also create an extensibility model on the conditions, in case other functionalities need to add additional information to a condition.
I didn't want to pollute the rule models with the compilation-specific fields, hence why I came up with these properties with a string-based key.

We can come up with an enum-based key, but then we won't be able to use the properties within' other nuget packages that extend the core package. Do you understand why I stood with the string in the end?

Reply:
I understand what you mean with the compile-focused (single assembly) instead of "system-focused".

I get that there would be no way we could define PropertyKeys in a specific package (other than the core), but can't really visualize why we would need that to happen...

But for the sake of flexibility I get the decision here 👌🏼 .

Reply:
That's basically it, respecting the open-closed principle here, that's it, we can't know what will come tomorrow 😄